### PR TITLE
Update Reth

### PR DESCRIPTION
**Pull request by 0xForerunner**
This was a pain haha. Everything seems to be working after the update though so we should be good to go.

### CODE REVIEW

File: world-chain-builder/Cargo.toml [hunk 1]
```diff
@@ -10,72 +10,76 @@ members = ["crates/*"]
...
+reth-chainspec = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-evm = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-db = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-provider = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-transaction-pool = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-payload-builder = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-network-api = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-node-api = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+reth-rpc = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
+# reth-rpc-types = { git = "https://github.com/paradigmxyz/reth", rev = "093b1a1" }
```
Comment (0xOsiris) [inline world-chain-builder/Cargo.toml]: Can we remove this if we aren't using it?

File: world-chain-builder/src/pool/validator.rs [hunk 2]
```diff
@@ -706,47 +702,10 @@ pub mod tests {
...
-        let inner = EthPooledTransaction::new(tx, 0);
-        let tx = WorldChainPooledTransaction {
-            inner,
-            semaphore_proof: Some(semaphore_proof.clone()),
-        };
-
-        validator.set_validated(&tx, &semaphore_proof).unwrap();
-    }
 
-    #[test]
```
Comment (0xOsiris) [inline world-chain-builder/src/pool/validator.rs:719]: Just curious the reason why we need to remove the deposit tx validation test?

Comment (0xForerunner) [reply world-chain-builder/src/pool/validator.rs:719]: It wasn't a very useful test. I can add it back if you think we want it!

File: world-chain-builder/Cargo.toml (hunk 1 above)
Comment (0xForerunner) [reply world-chain-builder/Cargo.toml]: Oh yeah I'll get rid of that.

**Pull request merged**